### ISSUE TITLE
Meaning of Vector3d results for hexagonal rotation axes / how to convert

### ISSUE THREAD

**Issue by maclariz** [pyxem/orix#358]
I have some nice hexagonal clustering data now.  And the misorientations at least partially make sense.  What I need now is to convert the vector3d results for the rotation axis into 4-vector representation. To do so, I need some clarity on the meaning of the 3 indices.  The first must be the a-axis, which really isn't controversial.  The third must be the c-axis (although is it scaled to a cartesian cubic representation?) and is the second the [01-10] axis and is it scaled?  Is there a conversion built in to 4-vectors?

Thanks

Ian

**Comment by hakonanes** [comment]
Hi Ian,

can you show how you obtained the misorientations? Are they the mean misorientations within each cluster as in [the misorientation clustering user guide](https://orix.readthedocs.io/en/stable/clustering_misorientations.html#Clustering)?

If the above is the case, the `cluster_mean.axis` (see user guide) should represent directions, as *unit* vectors (x, y, z), in the unit cell of the reference/initial orientation/grain in the *cartesian reference frame* (e1, e2, e3). To convert from (x, y, z) to [UVTW] in the *crystal reference frame* (a, b, c), you should be able to do the following:

```python
>>> from orix.crystal_map import Phase
>>> from orix.vector import Miller
>>> phase = Phase.from_cif("phase_file.cif")
>>> rot_axes = Miller(xyz=cluster_mean.axis, phase=phase)
>>> rot_axes.coordinate_format = "UVTW"
>>> rot_axes.round()
```

How the crystal reference frame (a, b, c) is aligned with the cartesian reference frame (e1, e2, e3) is explained in [this user guide](https://orix.readthedocs.io/en/stable/crystal_reference_frame.html).

**Comment by maclariz** [comment]
@hakonanes Yes, these are results from cluster analysis.  Actually slightly more clusters found than really exist (it split some areas of very similar misorientation).  And then I get all the misorientations to whatever I choose as the reference.  But this looks like it should work.  And should be able to use to identify relationships between different areas (there should be standard relationships in beta to alpha titanium transformation products).

**Comment by maclariz** [comment]
I already have a think in there for the point symmetry for HCP, and this is enough for the lattice vectors.  Why do I need to import the full cell?  Seems redundant.

**Comment by hakonanes** [comment]
>  Why do I need to import the full cell? Seems redundant.

Conversion from (x, y, z) in the cartesian reference frame to [uvw] in the crystal reference frame is done using the structure matrix, which requires the base lattice vectors and angles (see the end of the [crystal reference frame user guide](https://orix.readthedocs.io/en/stable/crystal_reference_frame.html#Converting-crystal-vectors)). But as you say, you don't need the full cell. Instead, you can create the `Phase` by hand by only passing the lattice parameters and the point group

```python
>>> from diffpy.structure import Lattice, Structure
>>> from orix.crystal_map import Phase
>>> from orix.quaternion.symmetry import D6h  # 6/mmm
>>> lattice = Lattice(1, 1, 1.5, 90, 90, 120)
>>> phase = Phase(structure=Structure(lattice=lattice), point_group=D6h)
>>> phase
<name: . space group: None. point group: 6/mmm. proper point group: 622. color: tab:blue>
>>> phase.structure.lattice.base
array([[ 1.       ,  0.       ,  0.       ],
       [-0.5      ,  0.8660254,  0.       ],
       [ 0.       ,  0.       ,  1.5      ]])
```

**Comment by maclariz** [comment]
so, I have this working.  

1

rot_axes

Miller (11,), point group 6/mmm, UVTW
[[ 0.1002  0.1228 -0.223   0.0315]
 [ 0.0792 -0.2149  0.1356 -0.0582]
 [-0.2259  0.11    0.1158 -0.0005]
 [ 0.2258 -0.1154 -0.1104 -0.0063]
 [ 0.      0.     -0.      0.    ]
 [-0.0897 -0.133   0.2228  0.0263]
 [-0.0731  0.214  -0.1408 -0.0576]
 [ 0.0598 -0.218   0.1582  0.0161]
 [ 0.1321 -0.2247  0.0927 -0.0018]
 [-0.1304  0.2249 -0.0946  0.0014]
 [ 0.2248 -0.1273 -0.0974 -0.0135]]
 
 But when I type rot_axes.data, I get the cartesian representation again.  Can I get these out in 4-vector form as pure numbers or a pure np array?

**Comment by maclariz** [comment]
I also get an error with:

rot_axes.round()

ValueError: The Miller-Bravais indices convention U + V + T = 0 is not satisfied

This suggests that the calculation is having rounding errors that it cannot recover from.  I checked the calculation and the t values are being calculated correctly.

**Comment by maclariz** [comment]
Normalisation is odd on conversion to UVTW format too.  All my vectors had length 0.27664721 after doing so (as calculated manually by me using my own knowledge of 4-vector maths).

**Comment by hakonanes** [comment]
Hm. So these are your `Misorientation.axes`

> Vector3d (11,)
[[ 0.4437 0.884 0.1473]
[ 0.3508 -0.8959 -0.2727]
[-0.9999 -0.0148 -0.0022]
[ 0.9995 -0.0127 -0.0297]
[ 0. 0. 0. ]
[-0.3972 -0.9094 0.123 ]
[-0.3237 0.9069 -0.2698]
[ 0.2648 -0.9614 0.0752]
[ 0.5847 -0.8112 -0.0085]
[-0.5772 0.8166 0.0065]
[ 0.9951 -0.0765 -0.063 ]]

and your misorientation symmetries are both 6/mmm (D6h)? If so, I'll try to troubleshoot tomorrow.

**Comment by maclariz** [comment]
@hakonanes I checked in the code, and no normalisation in conversion of uvw to UVTW, so that's why.  I am going to check your vector to uvw conversion tomorrow.  Just checking exam scripts now.

**Comment by hakonanes** [comment]
> no normalisation in conversion of uvw to UVTW, so that's why

For completeness, our conversion

https://github.com/pyxem/orix/blob/2d18672bd3beccf5b9da78e1f22e944d12870c33/orix/vector/miller.py#L826-L854

follows Marc De Graef's convention in *Introduction to conventional transmission electron microscopy* (2003).

>  I am going to check your vector to uvw conversion tomorrow

This conversion assumes the following crystal axes alignment: `e1 || a` and `e3 || c*`, and is implemented like so

https://github.com/pyxem/orix/blob/2d18672bd3beccf5b9da78e1f22e944d12870c33/orix/vector/miller.py#L789-L790

**Comment by maclariz** [comment]
@hakonanes Having checked exhaustively, all the main calculation seems to originate in diffpy (i.e. beyond your control) and the normalisation choices made there mean that your vectors once calculated to uvw or UVTW will always be shorter than one **a** vector of the lattice.  If I get the data out as numpy arrays, I can renormalise them using standard formulae for hexagonal indices.  No need to make any code changes, although making an option for a normalised version could be considered in future.

**Comment by maclariz** [comment]
The results once I fiddled with them start matching expectations from crystallographic theory, such as in Wang, Aindow and Starink
